metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala

```diff
@@ -18,19 +18,27 @@ class ConvertToNamedArguments(trees: Trees) extends CodeAction {
   import ConvertToNamedArguments._
   override val kind: String = l.CodeActionKind.RefactorRewrite
 
+  def getTermWithArgs(
+      apply: Term,
+      args: List[Tree],
+  ): Option[ApplyTermWithArgIndices] = {
+    val argIndices = args.zipWithIndex.collect {
+      case (arg, index)
+          if !arg.isInstanceOf[Term.Assign] && !arg
+            .isInstanceOf[Term.Block] =>
+        index
+    }
+    if (argIndices.isEmpty) firstApplyWithUnnamedArgs(apply.parent)
+    else Some(ApplyTermWithArgIndices(apply, argIndices))
+  }
   def firstApplyWithUnnamedArgs(
       term: Option[Tree]
   ): Option[ApplyTermWithArgIndices] = {
     term match {
       case Some(apply: Term.Apply) =>
-        val argIndices = apply.args.zipWithIndex.collect {
-          case (arg, index)
-              if !arg.isInstanceOf[Term.Assign] && !arg
-                .isInstanceOf[Term.Block] =>
-            index
-        }
-        if (argIndices.isEmpty) firstApplyWithUnnamedArgs(apply.parent)
-        else Some(ApplyTermWithArgIndices(apply, argIndices))
+        getTermWithArgs(apply, apply.args)
+      case Some(newAppl @ Term.New(init)) =>
+        getTermWithArgs(newAppl, init.argss.flatten)
```

Comment:
It doesn't work if the constructor has multiple parameters

```scala
class Foo(x: Int, y: Int)(z: Int)

new Foo(1, 2)(3)

// If we run the code action for `Foo(...)`
new Foo(1, 2)(z = 3)
```

Maybe we should treat differently for `Term.New`.

Reply:
I've encountered a lot of problems while working on case when the constructor has multiple parameter lists,
from finding which parameter list to extract, to getting correct type information (in scala I get things like `dotty.tools.dotc.core.Types$UnspecifiedErrorType$@6a25de0a`, in scala2 I get nothing :sweat_smile:), so for now this code actions will not be showing for `new` with multiple parameter lists

Reply:
There were some issues with finding which parameter list is selected (in cases like `Foo(1,2)<<(>>3)`) so, at least for now, we are converting all parameters for `Term.New`